pkg/addons/seaweedfs/upgrade.go

```diff
@@ -2,31 +2,43 @@ package seaweedfs
 
 import (
 	"context"
-	"log/slog"
 
 	"github.com/pkg/errors"
 	"github.com/replicatedhq/embedded-cluster/pkg/helm"
+	"github.com/sirupsen/logrus"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 func (s *SeaweedFS) Upgrade(ctx context.Context, kcli client.Client, hcli helm.Client, overrides []string) error {
+	if err := s.ensurePreRequisites(ctx, kcli); err != nil {
```

Comment:
i don't think this is idempotent. some it's functionality will fail if a resource already exists.

Reply:
I believe this pr makes it idempotent. Do you think otherwise?